src/Main/DetectiveGameMain.java

```diff
@@ -0,0 +1,183 @@
+                    return cases.get(choice - 1); // Return selected case
+                } else {
+                    System.out.println("Invalid choice. Please select a valid case number.");
+                }
+            } catch (NumberFormatException e) {
+                System.out.println("Invalid input. Please enter a number, 'add case', or 'quit'.");
+            }
+        }
+
+        // If the loop exits without returning, return null as a fallback
+        return null;
+    }
+
+    private static void handleAddCase(Scanner scanner, String input, String casesDir) {
+        Command addCaseCommand = CommandFactory.getCommand("add");
+        if (addCaseCommand == null) {
+            System.out.println("Error: Add case command not found.");
+            return;
+        }
+
+        if (input.equalsIgnoreCase("add case")) {
+            System.out.print("Enter the file path: ");
+            String filePath = scanner.nextLine().trim();
+            addCaseCommand.execute(new String[]{"add", "case", filePath}, null);
+        } else if (input.startsWith("add case ")) {
+            // Validate input length before extracting the file path
+            if (input.length() <= "add case ".length()) {
+                System.out.println("Error: No file path provided. Please specify a file path.");
+                return;
+            }
+            String filePath = input.substring("add case ".length()).trim();
+            addCaseCommand.execute(new String[]{"add", "case", filePath}, null);
+        } else {
+            System.out.println("Invalid input. Please type 'add case' or 'add case [file_path]'.");
+        }
+    }
+
+    private static void startGame(Scanner scanner, CaseFile caseFile, String casesDir) {
+        Building building = BuildingExtractor.loadBuilding(caseFile);
+        if (building == null) {
+            // If building is null, it means there were errors during case loading
+            return; // Exit to case selection menu
+        }
+
+        try {
+            SuspectExtractor.loadSuspects(caseFile, building);
+        } catch (IllegalStateException e) {
+            System.out.println(e.getMessage());
+            return; // Reload the case menu
+        }
+        GameObjectExtractor.loadObjects(caseFile, building);
+
+        TaskList taskList = new TaskList(caseFile.getTasks());
+        Detective detective = new Detective("Sherlock Holmes");
+        DoctorWatson watson = new DoctorWatson(caseFile.getWatsonHints());
+        GameContext context = new GameContext(building, detective, watson, new Journal(), taskList, caseFile);
+
+        watson.setCurrentRoom(building.getCurrentRoom());
+        building.setWatson(watson);
+
+        Letter letter = new Letter();
+        LetterExtractor.loadLetter(caseFile, letter);
+        letter.displayInvitation();
+        System.out.println("\nNow type 'start case' to begin the investigation.");
+
+        boolean isRunning = true;
+
+        while (isRunning) {
```

Comment:
Why not just `!context.isExitCurrentGame`? Context will not be reused for other games